### ISSUE TITLE
Introduce transformers

### ISSUE THREAD

**Issue by den-mentiei** [beardsvibe/buildfox#4]
```
transformer define: /D${param}

transformer options:
  no_exceptions = /EH-
  optimize = /Ox

# -----------

define = NDEBUG USE_LOGGER
filter variation:release
  options = no_exceptions optimize
```

**Comment by den-mentiei** [comment]
It seems that matching transformer is not needed at all - as we can just use simple variable setup (e.g. filtered by toolchain).

**Comment by den-mentiei** [comment]
Currently, transformer will treat it's parameter value as space separated list and then apply the specified substitution to each value with join afterwards.

Example:

```
transformer define: /D${param}
define = NDEBUG USE_LOGGER
```

will be transformed to:

```
define = /DNDEBUG /DUSE_LOGGER
```

Do we need to support space as a part of the list value? E.g. we can have `define = "WITH SPACE" WITHOUT_ONE`, but it 's not needed (am I right?), so I do not want to support it.

**Comment by jimon** [comment]
We can use ninja escaped sequences for this - escaped space :

```
define = WITH$ SPACE WITHOUT_ONE
```

I'm not 100% sure that space will cover all use cases, but it looks like a good start. Probably we will be able to add custom separator and custom jointer (symbol that joints token afterwards) to transformer definition later on.

**Comment by den-mentiei** [comment]
Sounds reasonable.

Also, we should support ninja escaped sequences inside the substitution pattern.

Example:

```
transformer test: $${param}:${param}
test = A B C
```

should be transformed to:

```
test = ${param}:A ${param}:B ${param}:C
```

**Comment by jimon** [comment]
Fixed in #6
